### PR TITLE
Fixes for lockup behavior outside Battle Mode

### PR DESCRIPTION
**Pull request by fdarosa2663**

### CODE REVIEW

File: props/saber_fett263_buttons.h [hunk 1]
```diff
@@ -5295,7 +5300,7 @@ SaberFett263Buttons() : PropBase() {}
           return true;
         }
 #endif
-        if (!battle_mode_ || swinging_) {
+        if ((!battle_mode_) || (battle_mode_ && swinging_)) {
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:5303]: This does exactly the same as what you had before, but less clearly.

File: props/saber_fett263_buttons.h [hunk 2]
```diff
@@ -1664,15 +1664,20 @@ SaberFett263Buttons() : PropBase() {}
...
-      } else {
-        SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
-        SaberBase::DoBeginLockup();
-        auto_lockup_on_ = true;
-        clash_type_ = CLASH_NONE;
-      }
+        if (clash_type_ == CLASH_BATTLE_MODE) {
+          if (SaberBase::GetClashStrength() < saved_gesture_control.clashdetect) {
+            SaberBase::DoClash();
+            clash_type_ = CLASH_NONE;
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:1670]: This line is the same in both branches, so it should probably go after the if-else.

File: props/saber_fett263_buttons.h (hunk 1 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h:5298]: It fixed a "hole", If it's not battle mode this should only do a clash ever, but if it is battle mode and the blade doesn't stop swinging then it should do a clash, the previous would let it through this check outside battle_mode and create a lockup if you had a clash without the swing stopping.  I added a comment to help clarify.

File: props/saber_fett263_buttons.h (hunk 2 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h]: consolidated

File: props/saber_fett263_buttons.h [hunk 3]
```diff
@@ -1664,15 +1664,20 @@ SaberFett263Buttons() : PropBase() {}
...
+          } else {
+            SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
+            SaberBase::DoBeginLockup();
+            auto_lockup_on_ = true;
+            clash_type_ = CLASH_NONE;
+          }
+	} else {
+          SaberBase::DoClash();
```
[the remark this answers is not in the document]
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h:1672]: fixed, thanks

File: props/saber_fett263_buttons.h (hunk 1 above)
Comment (profezzorn) [reply props/saber_fett263_buttons.h:5298]: Ok, time for logic tables:

B=battle_mode, S=swinging_, T=true, F=false

battle_mode:
__|_S__!S_
 B|   T    T 
!B|   F    F

swinging_:
__|_S__!S_
 B|   T    F 
!B|   T    F

!battle_mode_
__|_S__!S_
 B|   F    F 
!B|   T    T

!battle_mode || swinging_
__|_S__!S_
 B|   T    F 
!B|   T    T

battle_mode && swinging_
__|_S__!S_
 B|   T    F 
!B|   F    F

(!battle_mode) || (battle_mode_ && swinging_)
__|_S__!S_
 B|   T    F 
!B|   T    T

Assuming you get what the logic tables mean, you'll see that 
     (!battle_mode || swinging_)
and
     ((!battle_mode_) || (battle_mode_ && swinging_))
have exactly the same results in all cases, so it's not possible for this change to change any behavior, and it looked better before.

Comment (fdarosa2663) [reply props/saber_fett263_buttons.h]: Makes sense (or should have I guess), I reverted.

**Pull request merged**